postgres/postgres_test.go

```diff
@@ -1101,3 +1102,122 @@ func TestDisableCache(t *testing.T) {
 	}
 	os.Setenv("PREST_CACHE_ENABLE", "true")
 }
+
+func TestParseBatchInsertRequest(t *testing.T) {
+	config.Load()
+	Load()
+	m := make(map[string]interface{})
+	m["name"] = "prest"
+	m["pumpkin"] = "prest"
+	records := make([]map[string]interface{}, 0)
+	records = append(records, m)
+
+	var testCases = []struct {
+		description      string
+		body             []map[string]interface{}
+		expectedColNames []string
+		expectedValues   [][]interface{}
+		err              error
+	}{
+		{"first test",
+			records,
+			[]string{`"name"`, `"pumpkin"`},
+			[][]interface{}{{"prest", "prest"}},
+			nil},
+	}
+
+	for _, tc := range testCases {
+		t.Log(tc.description)
+		body, err := json.Marshal(tc.body)
+		if err != nil {
+			t.Errorf("expected no errors in http request, got %v", err)
+		}
+		req, err := http.NewRequest("POST", "/", bytes.NewReader(body))
+		if err != nil {
+			t.Errorf("expected no errors in http request, got %v", err)
+		}
+
+		colsNames, _, values, err := config.PrestConf.Adapter.ParseBatchInsertRequest(req)
+		if err != tc.err {
+			t.Errorf("expected errors %v in where by request, got %v", tc.err, err)
+		}
+
+		for _, name := range tc.expectedColNames {
+			if !strings.Contains(colsNames[0], name) {
+				t.Errorf("expected %#v in %#v, but wasn't!", tc.expectedColNames, colsNames)
+			}
+		}
+
+		if !reflect.DeepEqual(tc.expectedValues, values) {
+			t.Errorf("expected %v in %v", tc.expectedValues, values)
+		}
+	}
+}
+
+func TestBatchInsertSQL(t *testing.T) {
+	config.Load()
+	Load()
+	var testCases = []struct {
+		description  string
+		database     string
+		schema       string
+		table        string
+		names        []string
+		placeholders string
+		SQL          []string
+	}{
+		{
+			"first test",
+			"prest",
+			"public",
+			"test1",
+			[]string{`"id", "name"`, `"name", "id"`},
+			"$1, $2",
+			[]string{`INSERT INTO "prest"."public"."test1"("id", "name") VALUES($1, $2)`, `INSERT INTO "prest"."public"."test1"("name", "id") VALUES($1, $2)`},
```

Comment:
Why not use `insert` with multi value?
```
INSERT INTO table_name (field1, field2) VALUES (1,2), (3,4)
```

Reply:
@avelino @fabriziomello multi value a bit more difficult  because maps in Go do not have order. so we prefer use a loop as a first implementation